contracts/xasset-orchestrator/src/orchestrator.rs

```diff
@@ -0,0 +1,165 @@
+use crate::{error::Error, Contract};
+use loam_sdk::{
+    import_contract, loamstorage,
+    soroban_sdk::{
+        self, env, xdr::ToXdr, Address, Bytes, BytesN, Env, InstanceItem, Lazy, LoamKey,
+        PersistentMap, String, Symbol,
+    },
+    subcontract,
+};
+use loam_subcontract_core::Core;
+
+import_contract!(xasset);
+
+#[loamstorage]
+pub struct Storage {
+    /// XLM SAC contract address; initialized and then passed
+    /// to deployed xasset contracts
+    xlm_sac: InstanceItem<Address>,
+    /// XLM oracle contract, initialized and then passed
+    /// to deployed xasset contracts
+    xlm_contract: InstanceItem<Address>,
+    /// A map of deployed asset contracts to their asset symbol.
+    /// This is used to check if a contract is a valid asset contract
+    /// and to get the asset symbol from the contract address.
+    /// The key is the asset symbol, the value is the asset contract address.
+    assets: PersistentMap<String, Address>,
+}
+
+#[subcontract]
+pub trait IsOrchestratorTrait {
+    fn init(&mut self, xlm_sac: Address, xlm_contract: Address) -> Result<(), Error>;
+    #[allow(clippy::too_many_arguments)]
+    fn deploy_asset_contract(
+        &mut self,
+        asset_contract: Address,
+        pegged_asset: Symbol,
+        min_collat_ratio: u32,
+        name: String,
+        symbol: String,
+        decimals: u32,
+        annual_interest_rate: u32,
+    ) -> Result<Address, Error>;
+    fn get_asset_contract(&self, asset_symbol: String) -> Result<Address, Error>;
+    // Manually set an asset symbol to an existing contract address
+    fn set_asset_contract(
+        &mut self,
+        asset_symbol: String,
+        asset_contract: Address,
+    ) -> Result<(), Error>;
+    // Manually set an existing asset symbol to an existing contract address. Dangerous!
+    // This should only be used when needing to update an existing symbol's contract.
+    fn set_existing_asset_contract(
+        &mut self,
+        asset_symbol: String,
+        asset_contract: Address,
+    ) -> Result<(), Error>;
+}
+
+impl IsOrchestratorTrait for Storage {
+    fn init(&mut self, xlm_sac: Address, xlm_contract: Address) -> Result<(), Error> {
+        Contract::require_auth();
+        self.xlm_sac.set(&xlm_sac);
```

Comment:
Technically this can generated by a macro see here: https://github.com/loambuild/loam/pull/133

One annoying thing about this strategy is that it compiles with knowledge of which network the contract will be one. So perhaps we should add a `target/loam/testnet/...`. This way we can compile to multiple networks at once and then publish the correct one.

Reply:
I was wondering/asking if there was a code way to go get this! Hmm. I'm not sure there's a strong gain here vs having it be part of the deploy process

Reply:
Yeah ideally we would have `stellar registry publish/deploy` aware of the network and handle this. But passing is definitely fine for now and a little more straightforward with the other support.